File: diceroll.js
function diceRollSimulation(percobaan) {
  let results = new Array(20).fill(0);

  for (let i = 0; i < percobaan; i++) {
    let roll = Math.floor(Math.random() * 20) + 1;
    results[roll - 1]++;
  }

  for (let i = 0; i < results.length; i++) {
    console.log(`Sisi ${i + 1}: ${results[i]}/${percobaan}`);
    console.log(
      `Probabilitas Sisi ${i + 1}: ${(results[i] / percobaan).toFixed(6)}`
    );
  }
}

diceRollSimulation(1000000);

File: diceroll.py
import random

def dice_roll_simulation(percobaan):
    results = [0] * 20 

    for _ in range(percobaan):
        roll = random.randint(1, 20) 
        results[roll - 1] += 1

    for i in range(20):
        print(f"Sisi {i + 1}: {results[i]}/{percobaan}")
        print(f"Probabilitas Sisi {i + 1}: {results[i] / percobaan:.6f}")

dice_roll_simulation(1000000)
